java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -216,34 +137,38 @@ public ColumnVector(DType type, long rows, Optional<Long> nullCount,
    * @param contiguousBuffer the buffer that this is based off of.
    */
   private ColumnVector(long viewAddress, DeviceMemoryBuffer contiguousBuffer) {
+    super(viewAddress);
     offHeap = new OffHeapState(viewAddress, contiguousBuffer);
     MemoryCleaner.register(this, offHeap);
-    this.type = offHeap.getNativeType();
-    this.rows = offHeap.getNativeRowCount();
     // TODO we may want to ask for the null count anyways...
     this.nullCount = Optional.empty();
 
     this.refCount = 0;
     incRefCountInternal(true);
   }
 
-  static ColumnVector fromViewWithContiguousAllocation(long columnViewAddress, DeviceMemoryBuffer buffer) {
-    return new ColumnVector(columnViewAddress, buffer);
+  /** Creates a ColumnVector from a column view handle
+   * @return a new ColumnVector
+   */
+  public static ColumnVector fromColumnView(long columnViewHandle) {
```

Comment:
Why is this a handle and not an instance of `ColumnView`? I can see both having their place but the API I would expect most people to use would be walk down the column hierarchy until I reach what I want.  Now copy it out.  I would have the `ColumnView` class right there, why force the user to call another API to get the handle and also possibly lose all type safety while doing it.

Reply:
yes, I was passing what was minimally required and was torn on what the param type should be. Will fix.

Reply:
fixed.